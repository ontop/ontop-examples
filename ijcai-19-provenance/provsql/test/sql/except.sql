\set ECHO none
SET search_path TO public, provsql;

CREATE TABLE result_formula AS SELECT
  *, formula(provenance(),'personnel_name') FROM (
    (SELECT DISTINCT city FROM personnel)
  EXCEPT
    (SELECT p1.city                               
     FROM personnel p1, personnel p2
     WHERE p1.city = p2.city AND p1.id < p2.id
     GROUP BY p1.city
     ORDER BY p1.city)
  ) t
ORDER BY city;

SELECT remove_provenance('result_formula');
SELECT * FROM result_formula;

DROP TABLE result_formula;
