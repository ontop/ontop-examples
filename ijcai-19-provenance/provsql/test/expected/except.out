\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

   city   |                                         formula                                         
----------+-----------------------------------------------------------------------------------------
 Berlin   | ((Ellen ⊕ Susan) ⊖ (Ellen ⊗ Susan))
 New York | ((John ⊕ Paul) ⊖ (John ⊗ Paul))
 Paris    | ((Dave ⊕ Magdalen ⊕ Nancy) ⊖ ((Dave ⊗ Nancy) ⊕ (Dave ⊗ Magdalen) ⊕ (Magdalen ⊗ Nancy)))
(3 rows)

